### PR TITLE
Azure Monitor Exporter - Update Readme

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**

### CODE REVIEW

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 1]
```diff
@@ -1,34 +1,54 @@
 # Azure Monitor Exporter client library for .NET
 
-TODO
+The [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet) exporters which send [telemetry data](https://docs.microsoft.com/en-us/azure/azure-monitor/app/data-model) to the [Azure Monitor Platform](https://docs.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview) following the [OpenTelemetry Specification](https://github.com/open-telemetry/opentelemetry-specification).
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:3]: please remove en-us from doc links.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
 
+Install the Azure Monitor Exporter for OpenTelemetry .NET with NuGet:
+
+```shell
+dotnet add package OpenTelemetry.Exporter.AzureMonitor
+```
 
 ### Authenticate the client
 
+For the exporter to send data to Azure Monitor use a connection string, which is created automatically when creating an Application Insights resource.
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: this doesn't sound very right.. there is no authentication in this sdk.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 3]
```diff
@@ -1,34 +1,54 @@
...
 
 ## Key concepts
 
-TODO
+This exporter sends traces to the configured Azure Monitor Resource using HTTP. IP address used by the Azure Monitor is documented in [IP addresses used by Application Insights and Log Analytics](https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-addresses#outgoing-ports).
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:26]: HTTPS

Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:26]: address - plural form

Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:26]: remove en-us

Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:26]: i think this fall under pre-req. not key concepts

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 4]
```diff
@@ -1,34 +1,54 @@
...
 
 ## Examples
 
-TODO
+Refer to [`DemoTrace.cs`](tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/DemoTrace.cs) for a complete demo.
+
+```csharp
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:32]: add namespaces import also

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 5]
```diff
@@ -1,34 +1,54 @@
...
+OpenTelemetry.Sdk.CreateTracerProviderBuilder()
+    .SetResource(resource)
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:34]: resource - is coming from?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 2 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: I thought we use `InstrumentationKey` to authenticate on the server-side.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 6]
```diff
@@ -1,34 +1,54 @@
...
+    .AddSource("Samples.SampleClient")
+    .AddAzureMonitorTraceExporter(o => {
+        o.ConnectionString = $"InstrumentationKey=Ikey;";
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:37]: if we were asking to copy connectingstring, then they should be able to paste it as is here, right?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 4 above)
Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:32]: looks like not needed. ignore this.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 6 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:37]: Yes. You're correct.

**Review by cijothomas — APPROVED** [review]
suggestion provided, but can be addressed subsequently too.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 2 above)
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: We have instructions for fetching a connection string. Consider linking to this.
https://docs.microsoft.com/azure/azure-monitor/app/sdk-connection-string?tabs=net#finding-my-connection-string

Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: no, there is no authentication.
This is very different from most azure client sdks.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 6 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:37]: You can use this: `InstrumentationKey=00000000-0000-0000-0000-000000000000`. This is the minimal valid connection string, and this is the one I've used in all the documentation.

Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:37]: there is no mention of instrumentationkey ever in this doc, so the example should also avoid mentioning it.

`o.ConnectionString = "<actualconnectionstring>"`

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 3 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:26]: HTTP is a protocol. If needed we need to change it as HTTP over SSL.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 7]
```diff
@@ -1,34 +1,54 @@
...
+    })
+    .Build();
+```
 
 ## Troubleshooting
 
-TODO
+This exporter is fully instrumented for logging information at various levels of detail using the .NET EventSource to emit information. Logging is performed for each operation and follows the pattern of marking the starting point of the operation, it's completion, and any exceptions encountered. Additional information that may offer insight is also logged in the context of the associated operation.
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:44]: we can simply say, this sdk logs events to the EventSource names "xyz", and be listened to using any Eventlistener tools, like Perfview etc.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 2 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: Used this doc reference in prerequisites hence did not repeat it. Should we reuse it?

Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:22]: sorry, i missed that. you can disregard my comment :)

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 7 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:42]: Used the same pattern followed by other SDKs in Azure SDK repo

Comment (cijothomas) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:42]: Its correct for other SDKs in this repo, but not (yet) for this SDK.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporterEventSource.cs
There is no logging for each operation, completion, exception as claimed.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 8]
```diff
@@ -1,34 +1,53 @@
 # Azure Monitor Exporter client library for .NET
 
-TODO
+The [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet) exporters which send [telemetry data](https://docs.microsoft.com/azure/azure-monitor/app/data-model) to the [Azure Monitor Platform](https://docs.microsoft.com/azure/azure-monitor/app/app-insights-overview) following the [OpenTelemetry Specification](https://github.com/open-telemetry/opentelemetry-specification).
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:3]:
```suggestion
The [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet) exporters which send [telemetry data](https://docs.microsoft.com/azure/azure-monitor/app/data-model) to [Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/app/app-insights-overview) following the [OpenTelemetry Specification](https://github.com/open-telemetry/opentelemetry-specification).
```

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 9]
```diff
@@ -1,34 +1,53 @@
...
+dotnet add package OpenTelemetry.Exporter.AzureMonitor
+```
 ### Authenticate the client
 
+Exporter does not use authentication. 
 
 ## Key concepts
 
-TODO
+This exporter sends traces to the configured Azure Monitor Resource using HTTP. IP address used by the Azure Monitor is documented in [IP addresses used by Application Insights and Log Analytics](https://docs.microsoft.com/azure/azure-monitor/app/ip-addresses#outgoing-ports).
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:25]: Minor: probably worth to call out HTTP vs. HTTPS?

Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:25]:
```suggestion
This exporter sends traces to the configured Azure Monitor Resource using HTTP. IP addresses used by the Azure Monitor is documented in [IP addresses used by Application Insights and Log Analytics](https://docs.microsoft.com/azure/azure-monitor/app/ip-addresses#outgoing-ports).
```

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 10]
```diff
@@ -1,34 +1,53 @@
...
 
-TODO
+Refer to [`DemoTrace.cs`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.Demo.Tracing/DemoTrace.cs) for a complete demo.
+
+```csharp
+using OpenTelemetry.Exporter.AzureMonitor;
+using OpenTelemetry.Trace;
+
+OpenTelemetry.Sdk.CreateTracerProviderBuilder()
+    .AddSource("Samples.SampleClient")
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:36]: Minor: consider changing "sample" to "demo" as the word "sample" might get mixed with "sampling/sampler/sample".

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md [hunk 11]
```diff
@@ -1,34 +1,53 @@
...
+    .AddAzureMonitorTraceExporter(o => {
+        o.ConnectionString = "<actualconnectionstring>";
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md:38]: Nit: not sure if there is a general pattern to follow in the Azure SDK repo.

If not, consider `<Your Connection String>` just for better readability.

**Review by reyang — APPROVED** [review]
LGTM with couple non-blocking suggestions.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md (hunk 3 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/README.md]: I will get this revised later. Azure pipeline requires `Key concepts`, will figure out the way to remove this section.

**Pull request merged**